Plugin/Model/OrderRepositoryPlugin.php

```diff
@@ -0,0 +1,140 @@
+
+use Magento\Sales\Api\OrderRepositoryInterface;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\Data\OrderSearchResultInterface;
+use Magento\Sales\Api\Data\OrderExtensionFactory;
+use Magento\InventorySourceSelectionApi\Api\Data\SourceSelectionResultInterfaceFactory;
+use Ampersand\DisableStockReservation\Model\Sources as SourceModel;
+use Ampersand\DisableStockReservation\Service\SourcesConverter;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Ampersand\DisableStockReservation\Model\ResourceModel\Sources\CollectionFactory;
+use Ampersand\DisableStockReservation\Api\SourcesRepositoryInterface;
+
+/**
+ * Class OrderRepositoryPlugin
+ * @package Ampersand\DisableStockReservation\Plugin\Model
+ */
+class OrderRepositoryPlugin
+{
+    /**
+     * @var SourcesRepositoryInterface
+     */
+    private $sourcesRepository;
+
+    /**
+     * @var OrderExtensionFactory
+     */
+    private $orderExtensionFactory;
+
+    /**
+     * @var SourceSelectionResultInterfaceFactory
+     */
+    private $sourceSelectionResultInterfaceFactory;
+
+    /**
+     * @var SourcesConverter
+     */
+    private $sourcesConverter;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * OrderRepositoryPlugin constructor.
+     * @param SourcesRepositoryInterface $sourcesRepository
+     * @param OrderExtensionFactory $orderExtensionFactory
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultInterfaceFactory
+     * @param SourcesConverter $sourcesConverter
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        SourcesRepositoryInterface $sourcesRepository,
+        OrderExtensionFactory $orderExtensionFactory,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultInterfaceFactory,
+        SourcesConverter $sourcesConverter,
+        CollectionFactory $collectionFactory
+    ) {
+        $this->sourcesRepository = $sourcesRepository;
+        $this->orderExtensionFactory = $orderExtensionFactory;
+        $this->sourceSelectionResultInterfaceFactory = $sourceSelectionResultInterfaceFactory;
+        $this->sourcesConverter = $sourcesConverter;
+        $this->collectionFactory = $collectionFactory;
+    }
+
+    /**
+     * @param OrderRepositoryInterface $subject
+     * @param OrderInterface $result
+     * @return OrderInterface
+     */
+    public function afterGet(OrderRepositoryInterface $subject, OrderInterface $result): OrderInterface
+    {
+        try {
+            /** @var SourceModel $sourcesModel */
+            $sourcesModel = $this->sourcesRepository->getByOrderId($result->getId());
+
+            $sourceSelectionItems = $this->sourcesConverter
+                ->convertSourcesJsonToSourceSelectionItems($sourcesModel->getSources());
+
+            $this->applyExtensionAttributesToOrder($result, $sourceSelectionItems);
+        } catch (NoSuchEntityException $exception) {
+        }
```

Comment:
Should we add a comment inside the catch so we know why we're not doing anything with the error?